add exclusion setting for domains ( related issue #353 )

KMohZaid opened the pull request:
Problem in this PR is :  I am "noob" and i don't know how to use `i18n` translation . so we have to do it.

KMohZaid committed:
add exclusion setting for domains ( related issue - https://github.com/ClearURLs/Addon/issues/353 )

KMohZaid commented:
related issue : https://github.com/ClearURLs/Addon/issues/353

KMohZaid commented:
lol, i knew i should do something to keep 1 return url statement but lazy to do and it was feeling complex if i does

and sad, i forgot to remove test console.log

KMohZaid commented:
![image](https://github.com/user-attachments/assets/494d774a-1170-4a4e-a268-e2f7027c1d11)
preview

Beagle commented:
@KMohZaid +1 to have this merged in and released. I would love to finally have this feature in ClearURLs. It would make my life so much better.

KMohZaid commented:
Lol I didn't notice a mistake

In regex explanation, after `m` there was `?` which i think i accidentally removed

KMohZaid committed:
Fix mistake in regex domain exclude explanation

KMohZaid committed:
fix: 5 issues reported by sonarcloud
(https://github.com/ClearURLs/Addon/pull/383#issuecomment-2402006872)

1. using variable to store boolean and if any boolean is true, then
   return url
2. using for-of for list
3. we are using "storage" variable to store dictionary data not array,
   so made it "{}" from "[]"
4. "value" parameter in setData was string when case was "excludeDomains", still removed its console.log (not needed to console log it now so)

KMohZaid commented:
> ## [![Quality Gate Passed](https://camo.githubusercontent.com/f7c890f34e0efe8252e17e9afd5a85aa22fc9c81b42a0ec3d4821af9f55f250d/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f71672d7061737365642d323070782e706e67)](https://sonarcloud.io/dashboard?id=ClearURLs_Addon&pullRequest=383) **Quality Gate passed**
> 
> Issues ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0 New issues](https://sonarcloud.io/project/issues?id=ClearURLs_Addon&pullRequest=383&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) ![](https://camo.githubusercontent.com/86e5c40907bf9e8933fe6bcf185f7076622a76a3bef4b6d2b5fd583959409aef/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f61636365707465642d313670782e706e67) [0 Accepted issues](https://sonarcloud.io/project/issues?id=ClearURLs_Addon&pullRequest=383&issueStatuses=ACCEPTED)
> 
> Measures ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=ClearURLs_Addon&pullRequest=383&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=ClearURLs_Addon&pullRequest=383&metric=new_coverage&view=list) ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=ClearURLs_Addon&pullRequest=383&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=ClearURLs_Addon&pullRequest=383)

nice

tcelestino commented:
This PR is a feature to solves the #387 issue.

KMohZaid commented:
> This PR is a feature to solves the #387 issue.

No, this is just workaround. I think we should check what are tracking parameters for Google sheets domain and add rules for them. This feature will trust domain and allows unfiltered url loading

Neustradamus commented:
@KMohZaid: Can you create a PR on Gitlab too?
- https://gitlab.com/ClearURLs

If it is not merged, I think it is the moment to fork this project.

Linked to:
- https://github.com/ClearURLs/Addon/pull/372
